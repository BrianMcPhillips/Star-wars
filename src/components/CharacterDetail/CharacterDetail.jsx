import React from 'react';
import PropTypes from 'prop-types';

const CharacterDetail = ({ name, birth_year }) => (
  <div>
    <h1>{name}</h1>
    <h5>{birth_year}</h5>
  </div>
);

CharacterDetail.propTypes = {
  name: PropTypes.string.isRequired,
  birth_year: PropTypes.string.isRequired
};

export default CharacterDetail;
