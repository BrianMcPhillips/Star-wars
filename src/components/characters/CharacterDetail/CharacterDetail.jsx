import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const CharacterDetail = ({ 
  name, 
  birth_year, 
  hair_color, 
  films, 
  vehicles, 
  starships 
}) => (
  <div>
    <h1>{name}</h1>
    <h4>Birth Year: {birth_year}</h4>
    <h4>Hair color: {hair_color}</h4>
    <h4>Films: {films.map((film, i) => 
      <Link 
        to={`/film-detail/${film.split('films/')[1]}`} 
        key={film + i}>
        <p>{film}</p> 
      </Link>)}
    </h4>
    <h4>Vehicles: {vehicles.map((vehicle, i) => 
      <p key={vehicle + i}>{vehicle}</p>)}
    </h4>
    <h4>Starships: {starships.map((ship, i) => 
      <Link 
        to={`/ship-detail/${ship.split('ships/')[1]}`} 
        key={ship + i}>
        <p>{ship}</p>
      </Link>)}
    </h4>
  </div>
);

CharacterDetail.propTypes = {
  name: PropTypes.string.isRequired,
  birth_year: PropTypes.string.isRequired,
  hair_color: PropTypes.string.isRequired,
  films: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  vehicles: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  starships: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired
};

export default CharacterDetail;
