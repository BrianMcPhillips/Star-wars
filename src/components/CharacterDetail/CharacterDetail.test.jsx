import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterDetail from './CharacterDetail';

describe('CharacterDetail component', () => {
  afterEach(() => cleanup());
  it('renders CharacterDetail', () => {
    const { asFragment } = render(
      <CharacterDetail name="Luke Skywalker" birth_year="yo mama"/>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
