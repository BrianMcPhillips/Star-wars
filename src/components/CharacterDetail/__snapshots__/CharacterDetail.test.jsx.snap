// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetail component renders CharacterDetail 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Luke Skywalker
    </h1>
    <h5>
      yo mama
    </h5>
  </div>
</DocumentFragment>
`;
